### ISSUE TITLE
SKoda Enyaq. Power and Range are not updated

### ISSUE THREAD

**Issue by Marty56** [skodaconnect/homeassistant-myskoda#484]
I have noticed that the addon does not update the Charging Power and the Range while the car is being charged.
This bug occurs sometimes not always.

The old addon skoda connect does not show this issue

**Comment by WebSpider** [comment]
Please add some more details, such as version used, relevant entries in the logs, etc.

**Comment by WebSpider** [comment]
I have encountered this with the range sensor today myself, it does not update while travelling, using v1. 17.2 on HA2024.12 with an Enyaq on 3.7

Will check my logs when i get back to see if we are missing an update, or if the behaviour from MySkoda is simply different.

**Comment by gandalfb** [comment]
I am on 1.15.3

Enyaq 85 on Firmware 5.2

Battery Status is being updated while charging. 
All other entities are not updated. 

Connection and charge speed is not detected either. 

The app works and shows correct values. 

What is needed to provide insights? (Despite updating to latest version)

Thank you for the awesome plugin.

**Comment by WebSpider** [comment]
Can you see if this is solved in v1.18.1 ?

**Comment by gandalfb** [comment]
Currently loading and it still looks like it is behaving like described. 

Maybe the screenshots help.

![Screenshot_20250107-181441](https://github.com/user-attachments/assets/46d9f7fa-9c4c-4945-91d1-2116dc733dc0)
![Screenshot_20250107-181309](https://github.com/user-attachments/assets/d1fabf9f-9754-41e6-8f7a-d6bd43d68a28)
![Screenshot_20250107-181259](https://github.com/user-attachments/assets/3ab70c3a-efad-4617-8fbe-dcce1683d71c)
![Screenshot_20250107-181242](https://github.com/user-attachments/assets/fac5bf68-665b-4bad-b289-243b74af48b9)
![Screenshot_20250107-181221](https://github.com/user-attachments/assets/65b333f2-8b68-4621-99d1-94df5047e6a3)

**Comment by Krayster123** [comment]
Same happens here on my Enyaq. I noticed since the car was updated with the latest s/w version that the Skoda connect app itself has a limit in waking the car of 3 times a day found in the settings. If you exceed this it will stop updating car info and a message appears that it will not update because it wants to protect the battery. I've also tried to poll the car manually by updating the skoda4ever entity as described but I don't have this entity it seems.

**Comment by WebSpider** [comment]
> Same happens here on my Enyaq. I noticed since the car was updated with the latest s/w version that the Skoda connect app itself has a limit in waking the car of 3 times a day found in the settings. If you exceed this it will stop updating car info and a message appears that it will not update because it wants to protect the battery. I've also tried to poll the car manually by updating the skoda4ever entity as described but I don't have this entity it seems.

We never wake up the car, we on purpose did not develop that piece, in order to prevent problems like this.

Only issue is, that when you issue commands to change things in-car (charge limit, lock, etc), you will always need to wake the car, so skoda does that automatically.


With regard to the update entity: just use any entity that does come from your car. `skoda4ever` is there as an example

**Comment by Krayster123** [comment]
> > Same happens here on my Enyaq. I noticed since the car was updated with the latest s/w version that the Skoda connect app itself has a limit in waking the car of 3 times a day found in the settings. If you exceed this it will stop updating car info and a message appears that it will not update because it wants to protect the battery. I've also tried to poll the car manually by updating the skoda4ever entity as described but I don't have this entity it seems.
> 
> We never wake up the car, we on purpose did not develop that piece, in order to prevent problems like this.
> 
> Only issue is, that when you issue commands to change things in-car (charge limit, lock, etc), you will always need to wake the car, so skoda does that automatically.
> 
> With regard to the update entity: just use any entity that does come from your car. `skoda4ever` is there as an example

Thanks, makes more sense now. 
I've now looked again and the entities do update but only when there is a change. If there is no change in i.e. SoC when car is parked the entity shows the time when it received the last value. Also with a forced poll you don't see the timestamp from the entity change but without an changed actual update of the value it's not needed. Just looks like it lost connection.

**Comment by WebSpider** [comment]
I encountered this today on my Enyaq MY21 again, this time on HA2025.1.1 and integration 1.18.1. 
Next time I have a long charging session, I will enable debug logging and tracing to see what's going on here.

**Comment by WebSpider** [comment]
In the mean time I have charged a few times, and everything went fine 🤷 
Keeping this issue open, hoping someone catches the logs.

**Comment by gandalfb** [comment]
Just had the log. Now new version is out and installed :-)

Will set debug log again time loading. 

It looks to be connected to wake-up or sth. 

Availability was not great though of Skoda API itself. Had an App outage yesterday

**Comment by tigers75** [comment]
I have the same problem: some of the values (i.e. remaining time) are updated, others, like power and km/h charged are not:

![Image](https://github.com/user-attachments/assets/51af7980-26af-4c02-8ca6-0f32d9ab4382)

I have installed also the old Skoda Connect, app, which is still working and shows the correct values to I assume they are on the server correctly, since the other add-on can retrieve them:

![Image](https://github.com/user-attachments/assets/17a89ed0-0408-44ac-96b9-591e9160b7b7)

Oftentimes manual reload of MySkǒda updates the values, sometimes it doesn't.

This is on a MY21 Enyaq, upgraded to SW 3.7 by the dealer, it did work fine after the update when I first installed the MySkǒda add-on, HA2025.2.1, both addons updated to latest stable version (1.19.2 for MySkǒda. 1.2.13 for Skǒda connect).

**Comment by WebSpider** [comment]
This will be addressed in upcoming release.

**Comment by WebSpider** [comment]
I cannot reproduce this anymore on 1.20.0

**Comment by tigers75** [comment]
I'm on 1.20.0 and the issue is still there, never went away, and charging power is always stuck at 0.0, what can I do to help track the issue?

**Comment by tigers75** [comment]
This is what I'm experiencing: the upper graph is the old Skoda connect integration, the lower is the MySkoda (charging started at 23:00 and stopped at 23:47):

![Image](https://github.com/user-attachments/assets/3739d480-67dc-4d64-bbc6-0fe6d16a7a58)

The old integration is up to date and changes accordingly, while the new MySkoda is totally inconsistent: some values are updated every 15/20 minutes or so, other values are never right. I give an example here where all the sensors are from the MySkoda integration on latest version:

![Image](https://github.com/user-attachments/assets/61f9c6e5-a656-44f5-aacb-db2dbb4c791a)

Here we have, top to bottom (I translate to english form my italian version):

1. state of charging: it lags behind at the start going from unknown to charging after 10 minutes, than it's pretty much correct in switching to "ready to charge exactly when charge ends;
2. charging power: it gets a random update half an hour into charging, and another one half an hour after charge end;
3. battery SoC: is updated exactly when the sensor on line 1 is updated and then again exactly when charges end, it's not updated even when power from line 2 is;
4. remaining time: is is updated when sensor from line 1 is updated, again when power is updated but it's not when charge ends.

It all seems pretty random: some values are updated, some others not, at different times.
If more info is needed please ask, and I'll try to reproduce what's needed.
Also I'd kindly ask to reopen the issue.

**Comment by WebSpider** [comment]
Thanks for your feedback @tigers75 and @philippbirkholz !

Trying to summarize:

- @philippbirkholz mentions that SoC is updated frequently, but that charge power seems to be lagging and battery range lags even more
- @tigers75 mentions charge state lags behind about 10 minutes when starting the charge, charge power is updated inconsistently, SoC is charging inconsistently. Charge time is not updated at charge end

There are two things that may be relevant:
- Are you charging AC or DC? (maybe the messages of only one type of charging are not interpreted correctly)
- What is the polling interval of your integration?

**Comment by tigers75** [comment]
> * [@tigers75](https://github.com/tigers75) mentions charge state lags behind about 10 minutes when starting the charge, charge power is updated inconsistently, SoC is charging inconsistently. Charge time is not updated at charge end
> 
> There are two things that may be relevant:
> 
> * Are you charging AC or DC? (maybe the messages of only one type of charging are not interpreted correctly)
> * What is the polling interval of your integration?

Correct, but the lag is not constant at 10 minutes, in general the updates are all over the place.
I tested this only with AC charge at home.
I did not change the polling interval, so I guess it's at 30 minutes (but shouldn't this apply only to the car in standby mode and not when charging?)

**Comment by WebSpider** [comment]
> Correct, but the lag is not constant at 10 minutes, in general the updates are all over the place. I tested this only with AC charge at home. I did not change the polling interval, so I guess it's at 30 minutes (but shouldn't this apply only to the car in standby mode and not when charging?)

The reason this is relevant, is that a complete poll of the MySkoda servers is done whenever the car has not been updated for a period equal to the polling interval. Without debug+tracing it is hard to determine which messages coming from the car via push messages (mqtt) are the culprit here, so I am trying to rule out the complete poll.

The default is indeed 30 minutes.

It would help a lot if you could turn logging to debug, turn on tracing in the integration settings, and do a charging setting. Your logs will be a bit chatty, but will contain a lot of valuable information to see what's going on here.

**Comment by philippbirkholz** [comment]
@WebSpider thanks for reopening!

I'm experiencing a similar behavior with AC and DC charging and I also haven't set a custom polling interval. For me it looks like battery status is reported via push messages because during AC charging (7 kW and above) every single percentage increase is reported (through HA integration and MySkoda app). In the app it looks like also charging power and range is updated in the same interval, but this is not the case in HA.

I can try to get some logging during the next weeks for DC and single phase AC charging (with changing charging power). I will try to clean up the log files then a bit concerning other HA parts and personal data.

**Comment by philippbirkholz** [comment]
Here you can find the debug log from today for driving - AC charging - driving. During charging I once stopped and started charging again. Charging power looks reasonable but partly with significant delay and updated less often than SoC. Updated range is part of all SoC updates in the log, but not taken over to the HA sensor.

![Image](https://github.com/user-attachments/assets/c397518b-a406-4e43-a6c4-17169af52377)

config was:

> logger:
  default: info
  logs:
    myskoda: debug
    myskoda.mqtt: debug
    myskoda.rest_api: debug
    custom_components.myskoda.sensor: debug

Unfortunately I forgot to enable the API tracing, can create a log with tracing until end of next week.

[home-assistant_myskoda_2025-02-27_cleaned_up.log](https://github.com/user-attachments/files/19014963/home-assistant_myskoda_2025-02-27_cleaned_up.log)

**Comment by tigers75** [comment]
Looks like we may have different issues. Here's my log, hope I did it right: activated the debug log in the MyŠkoda integration setup, the filtered the system log with "myskoda" string. If I did it wrong, please guide me to get a more meaningful log.
(Note: I had a power outage tonight while charging, this may reflect in the logs).

[MySkoda.log](https://github.com/user-attachments/files/19037714/MySkoda.log)

**Comment by tigers75** [comment]
HI, I did a travel with a very long session with a granny charger at constant 2 kW, this is the result, comparing the MyŠkoda with the Skoda connect sensors. What I notice is that while the car is running both sensor are more or less the same and updated very frequently (see the middle graph where the yellow and blue line match pretty much exactly); while on charge the states are only roughly the same for the 2 integrations, with a noticeable shift of about 30 minutes in the MyŠkoda that looks like constantly lagging behind by 30 minutes.
I have no log but I noticed that while charging the power was not constantly updated, showing 0 kW even when there was constant 2 kW plugged in.

![Image](https://github.com/user-attachments/assets/841f8719-6186-4956-adea-099e1e22e86c)

**Comment by philippbirkholz** [comment]
Here now a log with API tracing.
[home-assistant_myskoda_2025-03-11_cleaned_up.log](https://github.com/user-attachments/files/19197956/home-assistant_myskoda_2025-03-11_cleaned_up.log)

Charging power was changing multiple times during the day between around 7, 5 and 2kW. This was reflected in the app, but not in the integration in HA. Same for charging rate (km/h), remaining charging time in contrast seems to update according to actual charging power.

![Image](https://github.com/user-attachments/assets/1ab19ad3-03fd-4d89-bf87-f1ac40375897)
![Image](https://github.com/user-attachments/assets/c37530f7-2d05-4ae7-9230-bdd360ebe52d)

**Comment by dvx76** [comment]
@philippbirkholz thanks for the detailed data

@WebSpider the first thing I notice in the last trace logs is that it starts at `2025-03-11 08:09:07` yet the first _scheduled_ update is only at `2025-03-11 16:02:44`, then every 30m until `2025-03-11 17:03:05` following by another gap until `2025-03-11 19:03:03`.

Meanwhile there are regular manual updates following reception of `change-soc` events.

What I think is happening is the (partial) manual update resets the poll interval for a (full) scheduled update. And indeed, at `2025-03-11 15:32:44` there is a manual update and then no more for 30m, resulting in the first scheduled update at `2025-03-11 16:02:44`!

This explains the observed behavior since the soc events only provide soc, range and remaining time.

**Comment by WebSpider** [comment]
> What I think is happening is the (partial) manual update resets the poll interval for a (full) scheduled update.
Yes, this is documented behaviour of the `DataUpdateCoordinator`

**Comment by WebSpider** [comment]
So i guess, we want to change what happens when we get a charging update here: Request a charging update, and update that part of the state as well, so all relevant sensors update?
